src/components/map/layers/RoutePathLinkLayer.tsx

```diff
@@ -14,33 +19,65 @@ interface RoutePathLinkLayerProps {
     weight: number;
 }
 
+@inject('popupStore')
+@observer
 export default class RoutePathLayer extends Component<RoutePathLinkLayerProps> {
 
     private onContextMenu = (routePathLinkId: string) => () => {
         this.props.onContextMenu(routePathLinkId);
     }
 
+    private renderRoutePathLinks() {
+        const routePathLinks = this.props.routePathLinks;
+
+        return routePathLinks.map((routePathLink, index) => {
+            return (
+                <Polyline
+                    positions={routePathLink.positions}
+                    key={index}
+                    color={this.props.color}
+                    weight={this.props.weight}
+                    opacity={this.props.opacity}
+                    onClick={this.props.onClick}
+                    onContextMenu={this.onContextMenu(routePathLink.id)}
+                />
+            );
+        });
+    }
+
+    private renderNodes() {
+        const routePathLinks = this.props.routePathLinks;
+
+        return routePathLinks.map((routePathLink, index) => {
+            return (
+                <>
+                    <NodeLayer
+                        key={index}
+                        node={routePathLink.startNode}
+                        isDisabled={routePathLink.startNodeType === NodeType.DISABLED}
+                        isTimeAlignmentStop={routePathLink.startNodeIsTimeAligned}
+                    />
+                    { index === routePathLinks.length - 1 &&
+                        <NodeLayer
+                            node={routePathLink.endNode}
+                            isDisabled={false}
+                            isTimeAlignmentStop={false}
```

Comment:
This data is only specified for start node on routelinks, and since this is the last link, we dont have the data if the node is disabled or time aligned. I suppose this means that end nodes cannot be either disabled or time aligned.

Reply:
Added a comment to the code